src/watchdog/src/watchdog.py

```diff
@@ -204,11 +204,11 @@ def _map_fn(item):
     map(_map_fn, podsJsonObject["items"])
 
 
-def collect_healthz(gauge, histogram, service_name, address, port, url):
+def collect_healthz(gauge, histogram, service_name, address, port, url, ca_path, headers):
     with histogram.time():
         error = "ok"
         try:
-            error = requests.get("http://{}:{}{}".format(address, port, url)).text
+            error = requests.get("https://{}:{}{}".format(address, port, url), headers = headers, verify = ca_path).text
```

Comment:
this assumes the k8s deployment will enable https. and this won't work with existing k8s deployment, right?